registry/client/transport/http_reader.go

```diff
@@ -163,10 +155,8 @@ func (hrs *httpReadSeeker) reader() (io.Reader, error) {
 	}
 
 	if hrs.readerOffset > 0 {
-		// TODO(stevvooe): Get this working correctly.
-
 		// If we are at different offset, issue a range request from there.
-		req.Header.Add("Range", "1-")
+		req.Header.Add("Range", "bytes="+strconv.FormatInt(hrs.readerOffset, 10)+"-")
```

Comment:
Changed to `fmt.Sprintf`. In general, I don't like format strings and try to avoid them. Too easy to get wrong, and no compile time checking.

Also, added a check that the response has status code 206 if there was a Range header in the request.